src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PageActionDescriptorChangeProvider.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.AspNetCore.Mvc.Razor.Internal;
+using Microsoft.AspNetCore.Razor.Evolution;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages.Internal
+{
+    public class PageActionDescriptorChangeProvider : IActionDescriptorChangeProvider
+    {
+        private readonly RazorProject _razorProject;
+
+        public PageActionDescriptorChangeProvider(RazorProject razorProject)
+        {
+            _razorProject = razorProject;
+        }
+
+        public IChangeToken GetChangeToken() => ((DefaultRazorProject)_razorProject).Watch("**/*.cshtml");
```

Comment:
Yep. this is why I wanted to limit us to watching a `Pages` folder, but that's beside the point. If it's bad we'll address it.